adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala

```diff
@@ -79,7 +79,7 @@ class AlignmentRecordConverter extends Serializable {
     // set canonically necessary fields
     builder.setReadName(adamRecord.getReadName.toString)
     builder.setReadString(adamRecord.getSequence)
-    builder.setBaseQualityString(adamRecord.getQual)
+    Option(adamRecord.getQual).fold(builder.setBaseQualityString("*"))(s => builder.setBaseQualityString(s))
```

Comment:
Simple pattern matching here would prevent double-setting the qualityString (unless I'm misreading this) and prevent allocating objects we won't use.